src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx

```diff
@@ -34,61 +32,24 @@ export type PupilViewsProgrammeListingProps = {
   baseSlug: string;
   yearSlug: PupilProgrammeListingData["yearSlug"];
   examboardSlug?: ExamboardData["examboardSlug"];
+  examboards: ExamboardData[];
+  tiers: TierData[];
 };
 
 export const PupilViewsProgrammeListing = ({
   programmes,
   baseSlug,
   yearSlug,
   examboardSlug,
+  examboards,
+  tiers,
 }: PupilViewsProgrammeListingProps) => {
-  const allExamboards: { [key: string]: ExamboardData[] } = _.groupBy(
-    getAvailableProgrammeFactor({
-      programmes,
-      factorPrefix: "examboard",
-    }) as ExamboardData[],
-    (examboard: ExamboardData) => examboard.examboard,
-  );
-
-  // This creates an array of examboards giving preference to non-legacy examboards
-  const examboards = Object.keys(allExamboards)
-    .map((examboard) => {
-      const mappedExamboard = allExamboards[examboard];
-      if (!Array.isArray(mappedExamboard) || mappedExamboard.length < 1) return;
-      return (
-        allExamboards[examboard]?.find(
-          (examboard: ExamboardData) => !examboard.isLegacy,
-        ) ?? mappedExamboard[0]
-      );
-    })
-    .filter((examboard): examboard is ExamboardData => examboard !== undefined);
-
```

Comment:
Moved to the helper function that is called in getStaticProps - options-pages-helpers.ts